s3_file_field/_multipart.py

```diff
@@ -179,6 +179,11 @@ def _iter_part_sizes(cls, file_size: int) -> Iterator[Tuple[int, int]]:
 
         # S3 multipart limits: https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
 
+        if file_size == 0:
```

Comment:
How does the existing algorithm fail to handle this case? Is it possible to make the existing code more robust to handle all cases, including length-0, without explicitly adding it as a special case?

Reply:
The relevant bit of the current algorithm looks like this:
```
remaining_file_size = file_size
while remaining_file_size > 0:
  this_part_size = ... # calculations
  yield this_part_number, this_part_size
  this_part_number += 1
```
If `file_size` was set to `0`, then the loop would never run, and nothing would be yielded. If we were to change the conditional to `while remaining_file_size >= 0`, we would yield `(part_number, 0)` infinitely when the loop would normally terminate after `remaining_file_size == 0`.

The only consistent solution I could come up was this one.